packages/components/src/focal-point-picker/focal-point.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PointerIconSVG,
+	PointerIconPathFill,
+	PointerIconPathOutline,
+	FocalPointWrapper,
+} from './styles/focal-point-style';
+
+export default function FocalPoint( {
+	coordinates = { left: '50%', top: '50%' },
+	isDragging = false,
+	...props
```

Comment:
I noticed you do that often (spread extra props) and I'm having trouble figuring out why or whether I consider this a good or bad practice. Are these needed? I know I personally try to avoid them if possible as they can lead to react warnings and can be hard to follow.

Reply:
For component libraries, prop spreading is common and often the default. The reason is because the consumer should be able to use HTML approved properties (like `aria-*` or `data-*`) just like they would on any HTML element. Not only that, but `on*` event callbacks should be available as well.

A (mostly) reliable solution for handling React warnings is that inner components often have prop filtering built in. In the case of components created with `@emotion/styled`, they filter out non React/HTML props before passing it down to elements like `<div />`, `<input />`, etc...

Reply:
> For component libraries, prop spreading is common and often the default. The reason is because the consumer should be able to use HTML approved properties (like aria-* or data-*) just like they would on any HTML element. Not only that, but on* event callbacks should be available as well.

When I use `FocalPointPicker` as a user, I don't know its internals whether it's a div, a button, or just a string... which makes me think, I don't know what HTML  prop are approved. This also makes it harder to be compatible with mobile. A block doing `< FocalPointPicker some-dom-specific-prop="something" />` won't translate properly to mobile. 

It makes me wonder if we should try to move away from these iteratively.